### PR TITLE
fix(build): use pinned hugo version

### PR DESCRIPTION
**Pull request by lidel**
This PR makes the same change as https://github.com/ipfs/website/pull/279 (make sure to check it as well) but for `blog.ipfs.io`:

- Automatically download and use safe version of `hugo` (v0.31.1) 
  - This fixes "no posts" error people have when building locally with Hugo >0.31.1
  - Makes local and remote build future-proof, enables us to migrate to new `hugo` without any pressure
- Bumped `nodemon` to fix event-stream issue
- Small cleanup of `README`, updated links and info that was no longer valid

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -82,15 +84,15 @@ static/
...
 
 ### Editing
 
 1. Make a change to a file
 2. Add and commit.
-3. `make build`
+3. `$ make build`
 4. `$ ipfs add -r build`
-  Only if you want a preview for other people (you can just use `make serve`). The path is `build`, in the website and the blog.
+  Only if you want a preview for other people. The path is `build`, in the website and the blog.
```
Comment (djdv) [inline README.md]: Suggestion:
``` md 
4. `$ ipfs add -r build` [Optional]
  You can add the "build" directory to IPFS to get a hash containing your version of [the blog](https://ipfs.io/ipfs/Qmc5GSKzJ8qYfaVivfLxKHxTBfUXzYqBmgomKhzycnbMJz/). 
```
But I'm not actually sure if this is a good way to word it or demonstrate how to take the hash and turn it into a public preview of the render.  ┐('～`；)┌

Comment (lidel) [reply README.md]: Hm.. this  may be scar tissue from time when blog and website were still a single thing.
I replaced the paragraph with simpler wording, without pointing at arbitrary CID: https://github.com/ipfs/blog/pull/199/commits/fed10b0093c2f815f9a90456faf70f2eafb8d04e

**Comment by olizilla** [comment]
Alas, the core reason for this PR, pinning the hugo version, got shipped in #209 

This PR has some README fixes in that are worth merging.

For the install content, the instructions should just be simplifed to get people using `make dev` for the hot-reloading dev experience, and `make` for the deployment version, as that is what all the other sites do. We should avoid duplicating the build process in both `make` and `npm`. For convinece, I'd have `npm start` run `make dev` but otherwise strip out all the others. I'll PR that.

**Comment by olizilla** [comment]
OK so, i've refreshed the README in #210

**Pull request closed without merging**